### ISSUE TITLE
Cancel reading of mailbox

### ISSUE THREAD

**Issue by RichiH** [neomutt/neomutt#248]
If I enter a large folder (<< 10k mails) by mistake, neomutt takes quite some time to read said mailbox. If I could just hit `^C` to stop reading that mailbox and go back into the previous view, that would save time every now and then.

**Comment by toogley** [comment]
I'm not sure, but `^G` could work. However, i don't know the exact difference between those two, i just found it helpful from time to time.

**Comment by flatcap** [comment]
I've just checked with 10K emails and Ctrl-G doesn't work either.

**Comment by guyzmo** [comment]
I would prefer `^G` to abort reading a large folder loading instead of `^C`, as it's what this keystroke is for: abort an operation gracefully, whereas `^C` is there to break things, and eventually get out of mutt.

**Comment by guyzmo** [comment]
Just pushed a patch that implements it for Maildir, mbox and Notmuch.

In that patch I decided to go for `^C` instead of `^G` (even though I prefered that one), because adding a `getch()` to the mail loading loop was doubling the load time, I tried different approaches (`timeout(1)`, `nodelay()`) but nothing was trivial enough to have an acceptable impact on mailbox loading.

When you hit `^C` it will abort the load, but still display all the mails up to the point you hit `^C`. If you hit `^C` again, it will then show you the yes/no exit menu.

* TODO:

- It's displaying an error message, but for some weird reasons, the screen gets refreshed just after the mailbox has been loaded, and thus the message appears for like half a second.
- Implementation for NNTP, POP and compress. As I'm not using those, I did not want to push untested code.
- Test mbox implementation (it should be 👌, but we'd better be sure).
